### PR TITLE
escm 1.1 (new formula)

### PR DESCRIPTION
**Pull request by kmizumar**
New formula for escm - Embedded Scheme Processor.

### CODE REVIEW

File: Library/Formula/escm.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    # `test do` will create, run in and delete a temporary directory.
+    #
+    # This test will fail and we won't accept that! It's enough to just replace
+    # "false" with the main program this formula installs, but it'd be nice if you
+    # were more thorough. Run the test with `brew test escm`. Options passed
+    # to `brew install` such as `--HEAD` also need to be provided to `brew test`.
+    #
+    # The installed folder is not in the path, so use the entire path to any
+    # executables being tested: `system "#{bin}/program", "do", "something"`.
+    system "false"
```
Comment (mikemcquaid) [inline Library/Formula/escm.rb]: Needs a test.

File: Library/Formula/escm.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  def install
+    # ENV.deparallelize  # if your formula fails when building in parallel
+
+    # Remove unrecognized options if warned by configure
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system "mkdir", "-p", "#{bin}"
```
Comment (mikemcquaid) [inline Library/Formula/escm.rb]: Use the `mkdir_p` function instead.

File: Library/Formula/escm.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+class Escm < Formula
+  homepage "http://practical-scheme.net/vault/escm.html"
+  url "http://practical-scheme.net/vault/escm-1.1.tar.gz"
+  sha1 "a1ddcbd88d9c26270f38c200abf5c8745c783faa"
+
+  # depends_on "cmake" => :build
```
Comment (mikemcquaid) [inline Library/Formula/escm.rb]: Remove all the generated comments.

**Comment by tdsmith** [comment]
This needs a scheme interpreter at runtime in normal use; do we want to try and express that as a Requirement somehow? Maybe we don't.

Can you add a test to this, @kmizumar? We can test it without installing a scheme interpreter using the preprocess-only `-E` option.

* in `install`, we can `prefix.install "escm.esc"`, and then
* in the test we can run `system bin/"escm", "-E", prefix/"escm.esc"`

The formula should also do `man1.install "escm.1"` in `install`.

File: Library/Formula/escm.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  homepage "http://practical-scheme.net/vault/escm.html"
+  url "http://practical-scheme.net/vault/escm-1.1.tar.gz"
+  sha1 "a1ddcbd88d9c26270f38c200abf5c8745c783faa"
+
+  def install
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    mkdir_p "#{bin}"
```
Comment (tdsmith) [inline Library/Formula/escm.rb]: `bin.mkpath` is a little more concise.

File: Library/Formula/escm.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/escm", "-v"
```
Comment (mikemcquaid) [inline Library/Formula/escm.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Comment by kmizumar** [comment]
Thank you so much for your comment, @tdsmith. Would it be possible for you to see if I understand your advice correctly? I have updated 324d69b.

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

**Comment by mikemcquaid** [comment]
A couple of final `audit` fixes needed: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/27018/version=yosemite/testReport/junit/brew-test-bot/yosemite/audit_escm___strict/

**Comment by DomT4** [comment]
Ping.

**Comment by DomT4** [comment]
Closing due to no response. Will welcome a new PR with the issues highlighted addressed. Thanks for your submission to Homebrew!

**Pull request closed without merging**